src/ostree/ot-admin-builtin-finalize-staged.c

```diff
@@ -51,10 +52,58 @@ ot_admin_builtin_finalize_staged (int argc, char **argv, OstreeCommandInvocation
   g_autoptr(GOptionContext) context = g_option_context_new ("");
   g_autoptr(OstreeSysroot) sysroot = NULL;
   if (!ostree_admin_option_context_parse (context, options, &argc, &argv,
-                                          OSTREE_ADMIN_BUILTIN_FLAG_SUPERUSER,
+                                          OSTREE_ADMIN_BUILTIN_FLAG_SUPERUSER | OSTREE_ADMIN_BUILTIN_FLAG_UNLOCKED,
                                           invocation, &sysroot, cancellable, error))
     return FALSE;
 
+  /* In case they're automounts, open /boot and the booted deployment on
+   * /sysroot so that the automounts don't expire. If they do expire and either
+   * gets unmounted, the service will be stopped.
+   */
+  int sysroot_fd = ostree_sysroot_get_fd (sysroot);
+  glnx_autofd int boot_fd = -1;
+  g_autoptr(OstreeDeployment) deployment = NULL;
+  g_autofree char *deployment_path = NULL;
+  glnx_autofd int deployment_fd = -1;
+  g_debug ("Opening /boot directory");
+  if (!glnx_opendirat (sysroot_fd, "boot", TRUE, &boot_fd, error))
+    return FALSE;
```

Comment:
OK as is, but perhaps note that `_ostree_sysroot_ensure_boot_fd()` exists; we'd just have to expose it via API (perhaps private) to use it.